runtime/src/main/java/com/dylibso/chicory/runtime/Memory.java

```diff
@@ -1,363 +1,106 @@
 package com.dylibso.chicory.runtime;
 
-import static com.dylibso.chicory.runtime.ConstantEvaluators.computeConstantValue;
-import static java.lang.Math.min;
-
-import com.dylibso.chicory.wasm.ChicoryException;
-import com.dylibso.chicory.wasm.UninstantiableException;
-import com.dylibso.chicory.wasm.types.ActiveDataSegment;
 import com.dylibso.chicory.wasm.types.DataSegment;
-import com.dylibso.chicory.wasm.types.MemoryLimits;
-import com.dylibso.chicory.wasm.types.PassiveDataSegment;
-import java.nio.BufferOverflowException;
-import java.nio.BufferUnderflowException;
-import java.nio.ByteBuffer;
-import java.nio.ByteOrder;
 import java.nio.charset.Charset;
-import java.nio.charset.StandardCharsets;
-import java.util.Arrays;
 
-/**
- * Represents the linear memory in the Wasm program. Can be shared
- * reference b/w the host and the guest.
- */
-public final class Memory {
+public interface Memory {
 
     /**
      * A WebAssembly page size is 64KiB = 65,536 bytes.
      */
-    public static final int PAGE_SIZE = 65536;
+    int PAGE_SIZE = 65536;
 
     /**
      * Maximum number of pages allowed by the runtime.
      * WASM supports 2^16 pages, but we must limit based on the maximum JVM array size.
```

Comment:
Is this actually a limit of `ByteBufferMemory`? With a different implementation of `Memory` that is backed by `MemorySegment`, segmented arrays, etc., are there any other places where `Integer.MAX_VALUE` would bite us?

Reply:
It's defined in the spec [here](https://webassembly.github.io/spec/core/exec/runtime.html#memory-instances)

![Screenshot from 2024-12-17 11-53-18](https://github.com/user-attachments/assets/3ddd9f12-3b26-4554-b9df-e96dfe6e8aec)